DOCS-5064 ASM compatibility info (#17583)
DataDog/documentation: content/en/security/application_security/enabling/compatibility/nodejs.md

---

> title: NodeJS Compatibility Requirements 

Reviewer: ```suggestion
title: Node.js Compatibility Requirements 
```

At the risk of opening a massive can of worms... have we standardized how we render "Node.js" (versus nodejs, NodeJS, etc.?) As far as I can tell, `Node.js` is how it's rendered by the developer (OpenJS Foundation) and the trademark is registered as such — [https://openjsf.org/wp-content/uploads/sites/84/2022/02/OpenJS-Foundation-Trademark-List-2022-02-14-1.pdf](https://openjsf.org/wp-content/uploads/sites/84/2022/02/OpenJS-Foundation-Trademark-List-2022-02-14-1.pdf)

kind: documentation
code_lang: nodejs
type: multi-code-lang
code_lang_weight: 50
---

## Language and framework compatibility

### Supported Node.js versions

> The Datadog Node.js library supports Node 14 and higher. In general, we follow the Node.js LTS policy. For more information, read the [APM documentation][1].

Reviewer: ```suggestion
The Datadog Node.js library supports Node 14 and higher. In general, Datadog follows the Node.js LTS policy. For more information, read the [APM documentation][1].
```

These are supported on the following architectures:

- Linux (GNU) x86-64
- Alpine Linux (musl) x86-64
- macOS (Darwin) x86-64
- Windows (msvc) x86, x86-64

You can monitor application security for Node.js apps running in Docker, Kubernetes, AWS ECS, and AWS Fargate.

### Supported frameworks

| Framework Web Server    | Minimum Framework Version   |
| ----------------------- | --------------------------- |
| Express                 | 4.0                         |

> <div class="alert alert-info">If you don't see your framework of choice listed, let us know! Fill out <a href="https://forms.gle/gHrxGQMEnAobukfn7">this short form to send us details</a>.</div>

Author: ```suggestion
<div class="alert alert-info">If you don't see your framework of choice listed, let us know! Fill out <a href="https://forms.gle/gHrxGQMEnAobukfn7">this short form to send details</a>.</div>
```

## ASM capabilities support

The following ASM capabilities are supported in the NodeJS library, for the specified tracer version:

| ASM capability                   | Minimum NodeJS tracer version |
| -------------------------------- | ----------------------------|
| Threat Detection <br/> --> Business logic API  | 3.31.1 <br/>   |

> | Threat Protection <br/> --> IP blocking <br/> --> Suspicious request blocking <br> --> User blocking   | 3.11.0<br/> --> 3.11.0<br/> --> not supported<br/> --> 3.11.0     |

Author: ```suggestion
| Threat Protection <br/> --> IP blocking <br/> --> Suspicious request blocking <br> --> User blocking   | <br/> --> 3.11.0<br/> --> not supported<br/> --> 3.11.0     |
```

> | Risk Management <br/> --> Third-party vulnerability detection <br/> --> Custom code vulnerability detection | 2.23.0 for NodeJS 12+, or 3.10.0 for NodeJS 14+ <br/>x.x<br/>x.x |

Author: ```suggestion
| Risk Management <br/> --> Third-party vulnerability detection <br/> --> Custom code vulnerability detection | 2.23.0 for NodeJS 12+, or 3.10.0 for NodeJS 14+ <br/><br/> |
```

> The minimum tracer version to get all supported ASM capabilities for NodeJS is 3.15.0.

Reviewer: ```suggestion
The minimum tracer version to get all supported ASM capabilities for Node.js is 3.15.0.
```

> <div class="alert alert-info">If you would like to see support added for any of the unsupported capabilities, let us know! Fill out <a href="https://forms.gle/gHrxGQMEnAobukfn7">this short form to send us details</a>.</div>

Author: ```suggestion
<div class="alert alert-info">If you would like to see support added for any of the unsupported capabilities, let us know! Fill out <a href="https://forms.gle/gHrxGQMEnAobukfn7">this short form to send details</a>.</div>
```

---
title: NodeJS Compatibility Requirements 
kind: documentation
code_lang: nodejs
type: multi-code-lang
code_lang_weight: 50
---

## Language and framework compatibility

### Supported Node.js versions

> The Datadog Node.js library supports the following Node.js versions:

Reviewer 2: Replace with--

The Datadog Node.js library supports Node 14 and higher. In general, we follow the Node.js LTS policy. For more information, please refer to the [APM documentation](https://docs.datadoghq.com/tracing/trace_collection/compatibility/nodejs/#nodejs-version-support)

> - Node.js 14 and higher

Reviewer 2: remove this line

These are supported on the following architectures:

- Linux (GNU) x86-64
- Alpine Linux (musl) x86-64
- macOS (Darwin) x86-64
- Windows (msvc) x86, x86-64

You can monitor application security for Node.js apps running in Docker, Kubernetes, AWS ECS, and AWS Fargate.

### Supported frameworks

| Framework Web Server    | Minimum Framework Version   |
| ----------------------- | --------------------------- |
| Express                 | 4.0                         |

<div class="alert alert-info">If you don't see your framework of choice listed, let us know! Fill out <a href="https://forms.gle/gHrxGQMEnAobukfn7">this short form to send us details</a>.</div>

## ASM capabilities support

The following ASM capabilities are supported in the NodeJS library, for the specified tracer version:

| ASM capability                   | Minimum NodeJS tracer version |
| -------------------------------- | ----------------------------|

> | Threat Detection <br/> --> Business logic API  | x.x <br/>x.x   |

Reviewer 2: for all threat detection: 3.13.1

Author: ```suggestion
| Threat Detection <br/> --> Business logic API  | 3.31.1 <br/>   |
```

> | Threat Protection <br/> --> IP blocking <br/> --> Suspicious request blocking <br> --> User blocking   | 3.11.0<br/>x.x<br/>not supported<br/>x.x     |

Reviewer 2: min. version 3.15.0.
NOT SUPPORTED: Suspicious Requests Blocking

| Risk Management <br/> --> Third-party vulnerability detection <br/> --> Custom code vulnerability detection | 2.23.0 for NodeJS 12+, or 3.10.0 for NodeJS 14+ <br/>x.x<br/>x.x |

> The minimum tracer version to get all supported ASM capabilities for NodeJS is y.y.

Reviewer 2: version: 3.15.0

Author: ```suggestion
The minimum tracer version to get all supported ASM capabilities for NodeJS is 3.15.0.
```
